pkg/api/v1/schema.go

```diff
@@ -35,8 +35,20 @@ type DomainSpec struct {
 }
 
 type Memory struct {
-	Value uint   `json:"value"`
-	Unit  string `json:"unit"`
+	// Allocated field defines the amount of memory a VM gets upon starting in kiloBytes
```

Comment:
https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-memory

_"Limits and requests for memory are measured in bytes. You can express memory as a plain integer or as a fixed-point integer using one of these suffixes: E, P, T, G, M, K. You can also use the power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki. For example, the following represent roughly the same value:"_

I kind of like the way kubernetes combines the unit and value into a single string. From a user's perspective it's really nice to be able to set one gigabyte using "1Gi" rather than having to translate that to kilobytes. 

Maybe we can support that method?

Reply:
If there is an API to convert that back to reasonable number type then yes. I need to use those numbers in calculations and the memory amount can be a pretty high number.

Reply:
actually I used it already, ParseQuantity is probably enough